sdk/docs/pages/typescript-sdk/cryptography/keypairs.mdx

```diff
@@ -1,109 +1,114 @@
-import { Callout } from 'nextra/components';
+# Key pairs 
 
-# KeyPairs
+The Sui TypeScript SDK provides `Keypair` classes that handle logic for signing and verification using the cryptographic key pairs
+associated with a Sui address. 
 
-KeyPair classes are provide logic for signing and verification using the cryptographic keypairs
-associated with an Address.
+The Sui TypeScript SDK supports three signing schemes:
 
-```typescript
-import { ED25519KeyPair } from '@mysten/sui.js/keypairs/ed25519';
+| Sign scheme | Class name | Import folder |
+| ----------- | ----- | --------------- |
+| Ed25519 | `Ed25519Keypair` | `@mysten/sui.js/keypairs/ed25519` |
+| ECDSA Secp256k1 | `Secp256k1Keypair` | `@mysten/sui.js/keypairs/ed25519` |
+| ECDSA Secp256r1 | `Secp256r1Keypair` | `@mysten/sui.js/keypairs/ed25519` | 
 
-// create a random KeyPair
-const keyPair = new ED25519KeyPair();
-// or create a KeyPair from an existing secret key (Uint8Array)
-const keyPair = ED25519KeyPair.fromSecretKey(secretKey);
+For information on these schemes, see the [Sui Wallet Specifications](https://docs.sui.io/testnet/learn/cryptography/sui-wallet-specs) topic.
 
-const publicKey = keyPair.getPublicKey();
-const message = new TextEncoder().encode('hello world');
+To use, import the key pair class your project uses from the `@mysten/sui.js/keypairs` folder. For example, to use the Ed25519 scheme, import the `Ed25519Keypair` class from `@mysten/sui.js/keypairs/ed25519`.
 
-const { signature } = await keyPair.signPersonalMessage(message);
-const isValid = await publicKey.verifyPersonalMessage(message, signature);
+```typescript
+import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519';
 ```
 
-## Public Keys
-
-Each KeyPair class has an associated PublicKey class. The PublicKey class that can be used to verify
-signatures, or get a the SuiAddress associated with the keyPair. PublicKeys can be access from a
-KeyPair, or constructed directly using the bytes (as a Uint8Array) of the PublicKey
+To create a random key pair (which identifies a Sui address), instantiate a new `Keypair` class. To reference a key pair from an existing secret key, pass the secret to the `fromSecretKey` function.
 
 ```typescript
-import { ED25519KeyPair, ED25519PublicKey } from '@mysten/sui.js/keypairs/ed25519';
+// random Keypair
+const keypair = new Ed25519Keypair();
+// Keypair from an existing secret key (Uint8Array)
+const keypair = Ed25519Keypair.fromSecretKey(secretKey);
+```
 
-const keyPair = new ED25519KeyPair();
-const bytes = keyPair.getPublicKey().toBytes();
+With your key pair created, you can use it to perform actions on the network. For example, you can use it to sign transactions, like the following code that creates and signs a personal message using the public key from the key pair created in the previous code:
```

Comment:
Keypairs will never act on the network.  They are purely running cryptography operations locally (in the future we may have keypairs that act as a proxy run the operations in a wallet or worker, but there won't be any interaction with the sui network

Reply:
Poorly worded on my part. I changed it to what I meant to write, let me know if it still sounds wrong.